### PR TITLE
Add support for multiple processes to the benchmark runner

### PR DESCRIPTION
**Pull request by fitzgen**
And a couple more little things

### CODE REVIEW

File: benchmarks-next/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+* Call `bench.start` exactly once during the execution of its `_start`
+  function. This is when the benchmark runner will start recording execution
+  time and performance counters.
+
+* Call `bench.end` exactly once during execution of its `_start` function, after
+  `bench.start` has already been called. This is when the benchmark runner will
+  stop recording execution time and performance counters.
+
+* Not use `stdout`. It may use `stderr` for logging, if necessary.
```
Comment (abrown) [inline benchmarks-next/README.md:43]: Why, though? If the reason is for us to be able to distinguish between sightglass output and benchmark output, we should explain that here.

File: benchmarks-next/blake3-scalar/rust-benchmark/src/main.rs [hunk 2]
```diff
@@ -11,13 +11,13 @@ extern "C" {
 
 fn main() {
     let buffer = [0 as u8; 64 << 10];
-    println!("[blake3] hashing a 64KB, zero-filled buffer");
+    eprintln!("[blake3] hashing a 64KB, zero-filled buffer");
```
Comment (abrown) [inline benchmarks-next/blake3-scalar/rust-benchmark/src/main.rs:14]: Where did the `benchmark.wasm` and `benchmark.wat` for this file go? Looks like they got deleted?

File: crates/artifact/src/docker.rs [hunk 3]
```diff
@@ -172,7 +172,7 @@ pub struct DockerId(String);
 
 impl From<String> for DockerId {
     fn from(s: String) -> Self {
-        DockerId(s.trim().to_string())
+        DockerId(s.trim().split_whitespace().last().unwrap().to_string())
```
Comment (abrown) [inline crates/artifact/src/docker.rs:175]: 👍 this is what I found was necessary to run on latest Docker (but not on my machine). This means we could likely add a CI task to do `benchmarks-next/build.sh` and then check if any benchmarks have changed to make sure the build is deterministic.

File: crates/cli/src/benchmark.rs [hunk 4]
```diff
@@ -1,19 +1,128 @@
...
+    #[structopt(long = "processes", default_value = "30", value_name = "PROCESSES")]
+    processes: usize,
+
+    /// How many samples should we take from a single process?
+    #[structopt(
+        long = "samples-per-process",
+        default_value = "1",
+        value_name = "SAMPLES_PER_PROCESS"
+    )]
+    samples_per_process: usize,
```
Comment (abrown) [inline crates/cli/src/benchmark.rs:36]: Some thoughts about this: 
 - should the in-process and multi-process benchmark commands be the same struct? (The in-process struct could use the same `Vec<_>` parameters for engines and Wasm files)
 - if they should be separate structs, should we call this one `MultiProcessBenchmarkCommand`?
 - how are we triggering this one? The other one corresponds to the `benchmark` command... is structopt somehow smart enough to see the extra parameters and choose this struct?
 - I didn't think we needed to spell out the `short = ...` and `long = ...` attributes?

**Review by abrown — APPROVED** [review]

File: crates/cli/src/benchmark.rs (hunk 4 above)
Comment (fitzgen) [reply crates/cli/src/benchmark.rs:36]: The `benchmark` subcommand is the multi-process runner now. The idea is that users basically won't use `in-process-benchmark`, just the regular `benchmark` subcommand will. We might even want to hide it from the help text.

I don't think we want to make the in-process benchmark commands the same. The struct definition contains some duplication, but that isn't a maintenance burden. There isn't any duplicated code, which is where things would get annoying. And I expect that as we flush out the benchmark runner, they will diverge even further.

File: crates/artifact/src/docker.rs (hunk 3 above)
Comment (fitzgen) [reply crates/artifact/src/docker.rs:175]: Yeah, getting this in CI would be great.

File: benchmarks-next/blake3-scalar/rust-benchmark/src/main.rs (hunk 2 above)
Comment (fitzgen) [reply benchmarks-next/blake3-scalar/rust-benchmark/src/main.rs:14]: Hmm... let me double check what happened here...

File: crates/cli/src/benchmark.rs (hunk 4 above)
Comment (abrown) [reply crates/cli/src/benchmark.rs:36]: I see the `#[structopt(name = "benchmark")]` on `InProcessBenchmarkCommand`?

File: benchmarks-next/blake3-scalar/rust-benchmark/src/main.rs (hunk 2 above)
Comment (fitzgen) [reply benchmarks-next/blake3-scalar/rust-benchmark/src/main.rs:14]: They are still there, AFAICT?

File: crates/cli/src/benchmark.rs (hunk 4 above)
Comment (fitzgen) [reply crates/cli/src/benchmark.rs:36]: Ah woops! That was a leftover and/or rebase issue. Funnily enough, it didn't seem to have an effect, as the command was still showing up as `in-process-benchmark`. Removing it now.

File: benchmarks-next/blake3-scalar/rust-benchmark/src/main.rs (hunk 2 above)
Comment (abrown) [reply benchmarks-next/blake3-scalar/rust-benchmark/src/main.rs:14]: Ok, I guess I misread the diff... must have been just a bunch of lines deleted.

File: crates/artifact/src/docker.rs (hunk 3 above)
Comment (abrown) [reply crates/artifact/src/docker.rs:175]: Let's do it as a part of #54

**Pull request merged**